cypress/integration/search/search_user_post.js

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/*eslint max-nested-callbacks: ["error", 3]*/
+
+import users from '../../fixtures/users.json';
+
+/**
+ * create new DM channel
+ * @param {String} text - DM channel name
+ */
+function createNewDMChannel(channelname) {
+    cy.get('#addDirectChannel').scrollIntoView().click();
+
+    cy.get('#selectItems').within(() => {
+        cy.get('input[type="text"]').scrollIntoView().type(channelname, {force: true});
+    });
+
+    // cy.get('div .clickable').children().contains(channelname).click({force: true});
+    cy.contains('.more-modal__description', channelname).click({force: true});
+    cy.get('#saveItems').click();
+}
+
+describe('Search in DMs', () => {
+    it('S14672 Search "in:[username]" returns results in DMs', () => {
+        // # Login and navigate to the app
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        const message = 'Hello';
```

Comment:
Can we make this a random message, something like?
```suggestion
        const message = 'Hello' + Date.now();
```